### PR TITLE
Redesign ERC721 to be align with recent ERC1155 update we made [ERC721] #3448

### LINKED ISSUE #3448: Redesign ERC721 to be align with recent ERC1155 update we made [ERC721]
The good news is, that we already made the layout for ERC1155 (at least on Android), which now can be applied to the ERC721 redesign. Note, that unlike ERC1155, in the Assets tab you will not find the menu icon in the top right. Since there no way for multiple selection, I skipped it. 

**Zeplin:**
https://zpl.io/bzzgow7

ERC721-ios

**Other Zeplin examples**
https://zpl.io/agEY0nR
https://zpl.io/2jK60A4
https://zpl.io/VK3RgX8

https://user-images.githubusercontent.com/51817359/142426556-8be43902-ea56-4741-b0c9-1f77eadca9db.mov

### LINKED ISSUE #3600: "Search for a token" search bar in Assets tab in ERC1155 (and soon ERC721) should filter
Currently does nothing.

Screenshot 2021-12-28 at 11 24 23 AM

### PR DESCRIPTION
**Pull request by oa-s**
Closes #3448, Closes #3600

### CODE REVIEW

**Comment by hboon** [comment]
Can tapping the right nav button to toggle between list and grid animate the layout transistion without much work? i.e. if it is using the same collection view already? Otherwise nevermind.

Screenshot 2021-12-28 at 11 24 23 AM

**Review by hboon — COMMENTED** [review]
```
pod 'IBPCollectionViewCompositionalLayout'
```

What does this do? Is it reasonable effort to not use it?

If we use it, is it possible to wrap it in a pod? Maybe wrapping `AutoLayoutCollectionView` in a pod and exposing just `AutoLayoutCollectionView`? Would it help reduce third party dependencies in the main app?

File: Podfile.lock [hunk 1]
```diff
@@ -311,6 +321,6 @@ SPEC CHECKSUMS:
   WalletConnectSwift: 046ae74e8cb69445be9c94c66ef66f9cb019b68f
   web3swift: 06118d4c4edc801444aaa995bbbddeda176b97ef
 
-PODFILE CHECKSUM: eed0b6581ff81af0f01f5be1e1134299396facb9
+PODFILE CHECKSUM: ff6d53cc41e2fbd275aa69e2b1f17e8bad6530c8
 
-COCOAPODS: 1.11.2
+COCOAPODS: 1.10.1
```
Comment (hboon) [inline Podfile.lock:326]: Need to run `bundle` and `bundle exec pod install` first to get the correct gem version.

**Comment by oa-s** [comment]
> pod 'IBPCollectionViewCompositionalLayout'
> What does this do? Is it reasonable effort to not use it?
> 
> If we use it, is it possible to wrap it in a pod? Maybe wrapping `AutoLayoutCollectionView` in a pod and exposing just `AutoLayoutCollectionView`? Would it help reduce third party dependencies in the main app?

It helps us to use improvance that were introduced in iOS 13 for UICollectionView, especially for building layouts, the AutoLayoutCollectionView actually its just experiment and it could be removed.

**Comment by hboon** [comment]
Remember to take note of #3632. The chain icons should appear when the token icons are small, and only when they are small

**Comment by oa-s** [comment]
@hboon the PR is ready for review

**Review by hboon — APPROVED** [review]
Merge when you are ready. The "WIP. Don't merge" tag is still applied.

**Comment by oa-s** [comment]
> The "WIP. Don't merge" tag is still applied.

i will double check.

**Pull request merged**